regression-tests/test-results/pure2-hashable.cpp

```diff
@@ -66,7 +66,7 @@ return ret;
 
 mystruct::mystruct(auto&& i_, auto&& j_, auto&& k_)
 requires (std::is_convertible_v<CPP2_TYPEOF(i_), std::add_const_t<cpp2::i32>&> && std::is_convertible_v<CPP2_TYPEOF(j_), std::add_const_t<std::string>&> && std::is_convertible_v<CPP2_TYPEOF(k_), std::add_const_t<cpp2::u64>&>) 
-                                                                                                       : base{ { 1 } }
+                                                                                                       : base{ 1 }
```

Comment:
The fact that `this: base = 1;`
now lowers to `: base{ { 1 } }`
in consistent with <https://github.com/hsutter/cppfront/issues/468#issuecomment-1627541323>
that I still haven't come to terms with:
> the idea that `x: quantity = 0;` is "explicit" (<https://github.com/hsutter/cppfront/issues/468#issuecomment-1627541323>).
> -- <https://github.com/hsutter/cppfront/issues/542#issuecomment-1650011244>

Reply:
Nah, it must be a bug that I don't omit the braces when the context already implies some.

Reply:
The problem is that `emit_special_member_function` centralizes the lowering of Cpp1 constructors and assignment operators.
<https://github.com/hsutter/cppfront/issues/844#issuecomment-1829023060> lists some of the bugs from that.
Maybe one of those PRs fixes this particular issue, so I'm leaving it as-is.